[BUG] run npm update --save results in error if dependency is defined with `file:`

### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

We have one of the local dependencies defined with `file:../dependency` Run of `npm update --save` results in 

```
npm ERR! Cannot read properties of null (reading 'isRoot')
```
and `package.json` is not updated

### Expected Behavior

`update` should skip `file:../` dependencies and update properly `package.json`

### Steps To Reproduce

just having one of dependencies defined as `file:../`  and run of `npm update --save` results in error

### Environment

- npm: 8.5.1
- Node.js: v16.14.0
- OS Name: MacOs
- System Model Name:
- npm config: nothing worth mentioning

--- a/workspaces/arborist/lib/arborist/reify.js
+++ b/workspaces/arborist/lib/arborist/reify.js
@@ -1308,7 +1308,7 @@ module.exports = cls => class Reifier extends cls {
     // to only names that are found in this list
     const retrieveUpdatedNodes = names => {
       const filterDirectDependencies = node =>
-        !node.isRoot && node.resolveParent.isRoot
+        !node.isRoot && node.resolveParent && node.resolveParent.isRoot
         && (!names || names.includes(node.name))
         && exactVersion(node) // skip update for exact ranges
 

--- a/workspaces/arborist/test/arborist/reify.js
+++ b/workspaces/arborist/test/arborist/reify.js
@@ -2622,5 +2622,33 @@ t.test('save package.json on update', t => {
       'should save no top level dep update to root package.json'
     )
   })
+
+  t.test('should not throw when trying to update a link dep', async t => {
+    const path = t.testdir({
+      one: {
+        'package.json': JSON.stringify({
+          name: 'one',
+          version: '1.0.0',
+          dependencies: {
+            two: 'file:../two',
+          },
+        }),
+      },
+      two: {
+        'package.json': JSON.stringify({
+          name: 'two',
+          version: '1.0.0',
+        }),
+      },
+    })
+
+    await t.resolves(reify(resolve(path, 'one'), { update: true, save: true, workspaces: [] }))
+
+    t.equal(
+      require(resolve(path, 'one', 'package.json')).dependencies.two,
+      'file:../two',
+      'should have made no changes'
+    )
+  })
   t.end()
 })

